src/effects/native/gatereffect.cpp

```diff
@@ -0,0 +1,236 @@
+#include "effects/native/gatereffect.h"
+
+namespace {
+    // Gain when the gate is open, higher than 1 to still have a decent level
+    constexpr double kMaxGain        = 1.5; 
+    // Attack slope
+    constexpr double kBaseAttackInc  = 0.001;
+    // Release slope
+    constexpr double kBaseReleaseInc = 0.0005;
+    }
+
+// static
+QString GaterEffect::getId() {
+    return "org.mixxx.effects.gater";
+}
+
+// static
+EffectManifest GaterEffect::getManifest() {
+    EffectManifest manifest;
+    manifest.setId(getId());
+    manifest.setName(QObject::tr("Gater"));
+    manifest.setAuthor("The Mixxx Team");
+    manifest.setVersion("1.0");
+    manifest.setDescription("");
```

Comment:
Please add a description for the effect and every parameter to describe what they do and how the scale of the parameter works. The descriptions appear as tooltips (when tooltips are enabled).

Reply:
Also add a shortName for the effect and every parameter, even if it is identical to the full name in English so it can be translated.